src/client/java/teammates/client/scripts/sql/DataMigrationForUsageStatisticsSql.java

```diff
@@ -0,0 +1,63 @@
+package teammates.client.scripts.sql;
+
+import java.util.UUID;
+
+import com.googlecode.objectify.cmd.Query;
+
+import teammates.storage.sqlentity.UsageStatistics;
+
+public class DataMigrationForUsageStatisticsSql extends
+    DataMigrationEntitiesBaseScriptSql<
+        teammates.storage.entity.UsageStatistics,
+        UsageStatistics> {
+
+    public static void main(String[] args) {
+        new DataMigrationForUsageStatisticsSql().doOperationRemotely();
+    }
+
+    @Override
+    protected Query<teammates.storage.entity.UsageStatistics> getFilterQuery() {
+        // returns all UsageStatistics entities
+        return ofy().load().type(teammates.storage.entity.UsageStatistics.class);
+    }
+
+    /**
+     * Set to true to preview the migration without actually performing it.
+     */
+    @Override
+    protected boolean isPreview() {
+        return true;
+    }
+
+    /**
+     * Always returns true, as the migration is needed for all entities from Datastore to CloudSQL .
+     */
+    @Override
+    protected boolean isMigrationNeeded(teammates.storage.entity.UsageStatistics entity) {
+        return true;
+    }
```

Comment:
Would it be possible to add a check: only returns true if the entity has not migrated yet? The other script does not have that check yet (only return true) as well

Reply:
Sure, the check will search the cloudSQL DB and check if another record with the same Id alr exists

Reply:
Hi @ziqing26, shall we figure out how to do this check as a team tomorrow? since it likely involves including some kind of HibernateUtil to check if the SQL side already has an entity with the same ID. 

We could implement the Hibernate accessing logic across the board in the main base script.

Reply:
Yep using HibernateUtil to check should be the idea.